src/bsp/tls/crypto-algorithms/xi_bsp_sha256.c

```diff
@@ -0,0 +1,34 @@
+/* Copyright (c) 2003-2017, LogMeIn, Inc. All rights reserved.
+ *
+ * This is part of the Xively C Client library,
+ * it is licensed under the BSD 3-Clause license.
+ */
+
+#include <xi_bsp_crypt.h>
+#include <xi_bsp_mem.h>
+#include <sha256.h>
+
+int xi_bsp_crypt_sha256_init( void** sha )
+{
+    *sha = xi_bsp_mem_alloc( sizeof( SHA256_CTX ) );
+
+    sha256_init( ( SHA256_CTX* )*sha );
+
+    return 0;
```

Comment:
OK, so yes, I forsee this living in a particular BSP implementation for a platform / product, but not in the BSP declarations and SDK that we provide. 

Also, don't we have a BSP Success error code that we could return instead of the magic number of 0?